core/app/models/spree/promotion/legacy_adjuster.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module Spree
+  class Promotion < Spree::Base
+
+    # This class encapsulates all the things the promotion system does to
+    # an order.
```

Comment:
Not mandatory, but a more detailed (high-level) explanation could help readers in understanding what's going on here without reading the whole class code.